intervals/main_objs.py

```diff
@@ -2146,8 +2195,260 @@ def presentationTypes(self, melodic_ngram_length=4, limit_to_entries=True,
+            print("Number of Non-Overlapping Voices:", non_overlaps)
+            # print the music
+            for c in range(1, count + 1):
+                music = tk.renderToSVG(c)
+                # display(SVG(music))
+                display(HTML(music))
+
+    # July 2022 Addition for printing hr types with Verovio
+    def verovioHomorhythm(self, homorhythm):
+
+        '''
+        This function is used to display the results of the homorhythm function
+        in the Notebook with Verovio.  Each excerpt follows the full measure
+        span of the homorhythm passage found by that function.
+
+        The function also displays metadata about each excerpt, drawn from the
+        homorhythm dataframe:  piece ID, composer, title, measure range,
+        and the minimum and maximum number of homorhythmic voices in that passage.
+
+        Usage:
+
+        First run the piece.homorhythm() function and assign it
+        to a new variable:
+
+        homorhythm = piece.homorhythm()
+
+        After any additional filtering, pass that variable to the print function:
+
+        piece.verovioHomorhythm(homorhythm)
+        '''
+
+
+
+        if self.path.startswith('Music_Files/'):
+            text_file = open(url, "r")
+            fetched_mei_string = text_file.read()
+        else:
+            response = requests.get(self.path)
+            fetched_mei_string = response.text
+        tk = verovio.toolkit()
+        tk.loadData(fetched_mei_string)
+        tk.setScale(30)
+        tk.setOption( "pageHeight", "1500" )
+        tk.setOption( "pageWidth", "2500" )
+
+        # Now get meas ranges and number of active voices
+        hr_list = list(homorhythm.index.get_level_values('Measure').tolist())
+        #Get the groupings of consecutive items
+        li = [list(item) for item in consecutive_groups(hr_list)]
+        final_list = []
+        new_final = []
+
+        # Look ahead and combine overlaps
+        for l in range(len(li)):
+        # look ahead
+            if l < len(li) - 1:
+                overlap_check = any(item in li[l] for item in li[l+1])
+                if overlap_check==False:
+                    sorted(li[l])
+                    final_list.append(li[l])
+                if overlap_check==True:
+                    combined = sorted(list(set(li[l] + li[l+1])))
+                    final_list.append(combined)
+        # Look back and combine overlaps
+        for l in range(len(final_list)):
+            new_final.append(final_list[0])
+            if l > 0:
+                overlap_check = any(item in final_list[l] for item in final_list[l-1])
+                if overlap_check==False:
+                    new_final.append(final_list[l])
+                if overlap_check==True:
+                    combined = sorted(list(set(final_list[l] + final_list[l-1])))
+                    new_final.append(combined)
+
+        # ensure final list is only unique lists
+        final_final = []
```

Comment:
Have you ever used a set? This is like a dictionary but just the keys, no values. Another way to think about it is a list that always only unique items. So instead of this for loop, it could just be:

final_final = set(new_final)

Even better would be to have made new_final a set to begin with, then you don't need to make final_final at all.

new_final = set()

Then instead of append, you use .add:

new_final.add(li[1])

This would also be a performance optimization.